src/wazuh_modules/inventory_harvester/tests/unit/fimInventoryDeleteAllEntries_test.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Wazuh inventory harvester
+ * Copyright (C) 2015, Wazuh Inc.
+ * February 20, 2025.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "common/clearElements.hpp"
+#include "mockFimContext.hpp"
+#include "mockIndexerConnector.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+class FimInventoryDeleteAllEntries : public ::testing::TestWithParam<MockFimContext::AffectedComponentType>
+{
+protected:
+    // LCOV_EXCL_START
+    FimInventoryDeleteAllEntries() = default;
+    ~FimInventoryDeleteAllEntries() override = default;
+    // LCOV_EXCL_STOP
+};
+
+class FimInventoryDeleteAllEntriesNoAgent : public ::testing::TestWithParam<MockFimContext::AffectedComponentType>
+{
+protected:
+    // LCOV_EXCL_START
+    FimInventoryDeleteAllEntriesNoAgent() = default;
+    ~FimInventoryDeleteAllEntriesNoAgent() override = default;
+    // LCOV_EXCL_STOP
+};
+
+TEST_P(FimInventoryDeleteAllEntries, ClearElementsDifferentComponents)
+{
+    std::map<MockFimContext::AffectedComponentType, std::unique_ptr<MockIndexerConnector>, std::less<>>
+        indexerConnectorInstancesFim;
+
+    auto component = GetParam();
+
+    // indexerConnectorInstances is not initialized with an invalid component.
+    if (component != MockFimContext::AffectedComponentType::Invalid)
+    {
+        indexerConnectorInstancesFim.emplace(component, std::make_unique<MockIndexerConnector>());
+    }
+
+    auto context = std::make_shared<MockFimContext>();
+    auto deleteElement =
+        std::make_shared<ClearElements<MockFimContext, MockIndexerConnector>>(indexerConnectorInstancesFim);
+
+    EXPECT_CALL(*context, agentId()).WillOnce(testing::Return("001"));
+    EXPECT_CALL(*context, affectedComponentType()).WillOnce(testing::Return(component));
+
+    if (!indexerConnectorInstancesFim.empty())
+    {
+        EXPECT_CALL(*indexerConnectorInstancesFim[component],
+                    publish("{\"id\":\"001\",\"operation\":\"DELETED_BY_QUERY\"}"))
+            .Times(1);
+        EXPECT_NO_THROW(deleteElement->handleRequest(context));
+    }
+    else
+    {
+        // Exception will be captured in inventoryHarvesterFacade
+        EXPECT_ANY_THROW(deleteElement->handleRequest(context));
+    }
+}
+
+TEST_P(FimInventoryDeleteAllEntriesNoAgent, ClearElementsDifferentComponentsNoAgent)
```

Comment:
Question: what is the purpose of this test?
The events should always have an ID, and if they haven't got one, the class isn't failing and it's sending a message that later will be rejected.

Reply:
I wanted to tests all cases. Since events with a missing agent id sent the message anyway, I checked the indexerConnector tests to add this case, but I found it was already covered.

Reply:
> the class isn't failing and it's sending a message that later will be rejected.
At that moment I was not sure that will happen, so I wanted to be sure with a test

Reply:
Great, thank you!